src/javaExercises/SortMatrix/SortMatrix.java

```diff
@@ -0,0 +1,85 @@
+package javaExercises.SortMatrix;
+
+import java.util.Arrays;
+import java.util.Random;
+
+public class SortMatrix {
+    public int[][] populateArrayWithRandomValues(int n, int max, int min) {
+
+        Random r = new Random();
+        int[][] arr = new int[n][n];
+        for (int i = 0; i < n; i++) {
+            for (int j = 0; j < n; j++) {
+                arr[i][j] = r.nextInt(max - min) - min + 1;
+            }
+        }
+        return arr;
+    }
+
+    public void printArray(int[][] arr) {
+
+        for (int[] value : arr) {
+            for (int value1 : value) {
+                System.out.print(value1 + "\t");
+            }
+            System.out.println();
+        }
+    }
+
+    public int[] sortLines(int[][] arr) {
```

Comment:
try using a Map<Integer,Integer[]>, were key will be the sum of elements per line and value will be the line from array, then you can just sort keys from map and get lines in sorted order of the keys

Reply:
Done